src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1473,6 +1504,30 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
                 return solution;
             }
 
+            internal Solution UpdateProjectDocumentsOrderForBatch(Solution solution, ProjectId projectId)
+            {
+                var multipleOrderedFilePaths = _multipleOrderedFilePathsInBatch.ToImmutableArray();
+
+                _multipleOrderedFilePathsInBatch.Clear();
+
+                // The action of ordering documents was not included in the batch. Simply return the solution.
+                if (multipleOrderedFilePaths.IsEmpty)
+                {
+                    return solution;
+                }
+
+                // We only care about the latest file paths.
+                return UpdateProjectDocumentsOrder(solution, projectId, multipleOrderedFilePaths.Last());
```

Comment:
Why did we hold onto all of that to only use the .Last() in the end? If there's multiple reorders, should we just clear out the old ones?

Reply:
I'll change it to something a bit more sensible.